### PR TITLE
MBS-9714 Copy button in documentation

### PR DESCRIPTION
**Pull request by eugene-krivobokov**
Peek 2020-10-11 19-08

### CODE REVIEW

File: docs/content/docs/contributing/Docs.md [hunk 1]
```diff
@@ -289,18 +289,6 @@ Markdown content
 Markdown content
 {{< /details >}}
 
-### Expand
```
Comment (eugene-krivobokov) [inline docs/content/docs/contributing/Docs.md:292]: Deprecated shortcode in favor of `details`

File: docs/layouts/partials/docs/inject/head.html [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        }
+        div.highlight:hover .copy-code-button,
+        div.highlight:hover .copy-code-button:focus {
+            opacity: 1;
+            filter: invert(100%);
+        }
+        div.highlight:hover .copy-code-button:hover,
+        div.no-highlight:hover .copy-code-button:hover {
+            opacity: 1;
+            filter: invert(20%) sepia(85%) saturate(2162%) hue-rotate(201deg) brightness(100%) contrast(91%);
```
Comment (eugene-krivobokov) [inline docs/layouts/partials/docs/inject/head.html:42]: Nice calculator:  https://codepen.io/sosuke/pen/Pjoqqp

File: docs/layouts/partials/docs/inject/head.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
+<!-- This is a partial template (https://gohugo.io/templates/partials/) to inject into hugo-book theme header -->
+<!-- It adds a copy button to code blocks -->
+
+{{ if (findRE "<pre" .Content 1) }}
```
Comment (eugene-krivobokov) [inline docs/layouts/partials/docs/inject/head.html:4]: https://gohugo.io/functions/findRE/

File: docs/layouts/partials/docs/inject/head.html [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    <script defer src="https://unpkg.com/clipboard@2/dist/clipboard.min.js">
```
Comment (eugene-krivobokov) [inline docs/layouts/partials/docs/inject/head.html:5]: https://clipboardjs.com/

File: docs/layouts/shortcodes/mermaid.html [hunk 5]
```diff
@@ -8,7 +8,7 @@
 - Use CDN instead of inlining megabytes of js in repository
 ======
 -->
-<script src="https://unpkg.com/mermaid@8.8.0/dist/mermaid.min.js" type="text/javascript" async></script>
+<script defer src="https://unpkg.com/mermaid@8.8.2/dist/mermaid.min.js" type="text/javascript"></script>
```
Comment (eugene-krivobokov) [inline docs/layouts/shortcodes/mermaid.html:11]: https://flaviocopes.com/javascript-async-defer/#with-async-in-the-head

File: docs/layouts/partials/docs/inject/head.html (hunk 2 above)
Comment (eugene-krivobokov) [reply docs/layouts/partials/docs/inject/head.html:42]: The dark theme is not supported yet. Maybe sometime

File: docs/layouts/partials/docs/inject/head.html [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+        document.querySelectorAll('pre > code').forEach(function (codeBlock) {
+            var button = document.createElement('button');
+            button.className = 'copy-code-button';
+            button.type = 'button';
+            button.title = 'Copy'
+
+            var buttonImage = document.createElement('svg');
+            buttonImage.setAttribute('viewBox', '0 0 24 24')
+            buttonImage.innerHTML = '{{ "/static/copy-24.svg" | readFile }}'
```
Comment (eugene-krivobokov) [inline docs/layouts/partials/docs/inject/head.html:65]: Need to inline SVG to filter colors

**Review by dsvoronin — APPROVED** [review]

File: docs/layouts/partials/docs/inject/head.html [hunk 7]
```diff
@@ -0,0 +1,99 @@
+<!-- This is a partial template (https://gohugo.io/templates/partials/) to inject into hugo-book theme header -->
```
Comment (sergioboi) [inline docs/layouts/partials/docs/inject/head.html:1]: Copy-paste?

Comment (eugene-krivobokov) [reply docs/layouts/partials/docs/inject/head.html:1]: I didn't get it. Copy-paste of what?
This is a comment for readers. We didn't use partials before. This is why I decided to put this hint.

Comment (sergioboi) [reply docs/layouts/partials/docs/inject/head.html:1]: The whole html file. Did you write it or copy-paste from somewhere? Because It looks HARD for me

Comment (eugene-krivobokov) [reply docs/layouts/partials/docs/inject/head.html:1]: This is not a copy-pasted implementation. The idea is based on this article: https://www.dannyguo.com/blog/how-to-add-copy-to-clipboard-buttons-to-code-blocks-in-hugo/ 

I propose not to pay too much attention to details of implementation. I think of it as a temporary solution before  https://github.com/alex-shpak/hugo-book/issues/281

The other option - migrate to mkdocs. It has this feature.

Comment (sergioboi) [reply docs/layouts/partials/docs/inject/head.html:1]: I understand

**Pull request merged**